src/arm/debugger/memory-debugger.c

```diff
@@ -92,12 +92,13 @@ CREATE_MULTIPLE_WATCHPOINT_SHIM(storeMultiple, WATCHPOINT_WRITE)
 CREATE_SHIM(setActiveRegion, void, (struct ARMCore* cpu, uint32_t address), address)
 
 static bool _checkWatchpoints(struct ARMDebugger* debugger, uint32_t address, struct mDebuggerEntryInfo* info, enum mWatchpointType type, uint32_t newValue, int width) {
-	--width;
 	struct mWatchpoint* watchpoint;
 	size_t i;
+	uint32_t minAddress = address / width * width;
```

Comment:
oh god no division in hot code

It's always going to be a power of 2, so you can -1 to get the mask:

```C
address & ~(width - 1)
```